### PR TITLE
Implemented game server pinging

### PR DESCRIPTION
**Pull request by Laugslander**

### CODE REVIEW

File: Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java [hunk 1]
```diff
@@ -73,12 +81,54 @@ public MainServerController() {
...
+    /**
+     * Continously pings all game servers to check if the RMI connection is
+     * active. A random value is sent to the game servers for them to process
+     * with a pre-defined formula. If the result of the formula is not correct,
+     * the connected game server is possibly rogue. If the result of the formula
+     * is not correct or there was no response at all, the game server is
+     * deregistered.
+     */
+    public void continuouslyPingGameServers() {
+        timer.scheduleAtFixedRate(new TimerTask() {
```
Comment (TehMessiah) [inline Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java]: MAJOR Reduce this anonymous class number of lines from 27 to at most 20, or make it a named class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1188)

File: Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java [hunk 2]
```diff
@@ -73,12 +81,54 @@ public MainServerController() {
...
+
+                            LOGGER.log(Level.WARNING, "A game server is deregistered because it is no longer active.", e);
+                        } catch (RogueGameServerException e) {
+                            serversToRemove.add(server);
+
+                            LOGGER.log(Level.WARNING, "A game server is deregisterd because it did not return the correct result of the predefined ping formula.", e);
+                        }
+                    });
+
+                    serversToRemove.stream().forEach(s -> deregisterGameServer(s));
```
Comment (TehMessiah) [inline Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java]: MINOR Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1612)

**Comment by TehMessiah** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java (hunk 2 above)
Comment (BeeHiveJava) [inline Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java]: Hij heeft gelijk! `forEach(MainServerController::deregisterGameServer)`

**Review by BeeHiveJava — CHANGES_REQUESTED** [review]

File: Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java [hunk 3]
```diff
@@ -73,12 +81,61 @@ public MainServerController() {
...
+
+                    LOGGER.log(Level.WARNING, "A game server is deregistered because it is no longer active.", e);
+                } catch (RogueGameServerException e) {
+                    serversToRemove.add(server);
+
+                    LOGGER.log(Level.WARNING, "A game server is deregisterd because it did not return the correct result of the predefined ping formula.", e);
+                }
+            });
+
+            serversToRemove.stream().forEach(s -> deregisterGameServer(s));
```
Comment (ghost) [inline Soccar [Main-Server]/src/nl/soccar/mainserver/rmi/MainServerController.java:129]: MINOR Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AS1612)

**Comment by ghost** [comment]
SonarQube analysis reported 1 issue
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Review by BeeHiveJava — APPROVED** [review]

**Pull request merged**